package/MDAnalysis/coordinates/PDB.py

```diff
@@ -678,6 +682,10 @@ def _check_pdb_coordinates(self):
             except OSError as err:
                 if err.errno == errno.ENOENT:
                     pass
+            except TypeError:
+                if isinstance(self.filename, StringIO):
+                    pass
```

Comment:
This is equivalent to

```python
except TypeError:
    pass
```

Shouldn't it be the following?

```python
except TypeError:
    if isinstance(self.filename, StringIO):
        pass
    else:
        raise
```

Reply:
Definitely! Thanks for spotting this problem!

Reply:
The same logic should apply to the previous exception as well, which is suffering from the same problem. 

However, this rise a question: should we ignore all the exceptions that can occur while trying `os.remove` so that the correct exception `ValueError` (which is the problem triggering `os.remove`) is thrown instead?

Reply:
We can at least wonder what would be the equivalent of an `os.remove` on a `StringIO`. Now, we have an odd discrepancy in behaviour between files and StringIO. Maybe it should be a different issue?

Reply:
I think the equivalent of `os.remove` for `StringIO` is just getting out of scope. However, it looks like that calling `close()` already frees the memory buffer:
> **StringIO.close()**
Free the memory buffer. Attempting to do further operations with a closed StringIO object will raise a ValueError.

Reply:
I guess, then, that the handling is appropriate.